koji-scm-lib/src/main/java/hudson/plugins/scm/koji/client/BuildMatcher.java

```diff
@@ -68,128 +67,107 @@ public BuildMatcher(String currentURL, Predicate<String> notProcessedNvrPredicat
     }
 
     public Optional<Build> getResult() {
-        Stream<Build> results = listMatchingBuilds();
-
-        Optional<Build> buildOpt = results
-                .filter(b -> notProcessedNvrPredicate.test(b.getNvr()))
+       return listMatchingBuilds().stream()
+                .filter(build -> notProcessedNvrPredicate.test(build.getNvr()))
                 .findFirst();
-        return buildOpt;
     }
 
-    public Object[] getAll() {
-        Stream<Build> results = listMatchingBuilds();
-
-        Object[] buildOpt = results
-                .filter(b -> notProcessedNvrPredicate.test(b.getNvr()))
-                .toArray();
-        return buildOpt;
+    public Build[] getAll() {
+        return listMatchingBuilds().stream()
+                .filter(build -> notProcessedNvrPredicate.test(build.getNvr()))
+                .toArray(Build[]::new);
     }
 
-    private Stream<Build> listMatchingBuilds() {
-        List<Object> builds = Arrays.stream(pkgName.split(" "))
-                .map(this::listPackageBuilds)
-                .flatMap(Collection::stream)
-                .collect(Collectors.toList());
+    private List<Build> listMatchingBuilds() {
+    	final List<Build> builds = new ArrayList<>();
+    	for (String packageName : pkgName.split(" ")) {
+    		builds.addAll(listPackageBuilds(packageName));
```

Comment:
strange indentation. Please setup your IDE for this project to match the project conventions (I think it's 4 spaces) and reformat changed files. You have tabs in many places.